src/Header.jsx

```diff
@@ -32,14 +33,12 @@ subscribe(APP_CONFIG_INITIALIZED, () => {
     MINIMAL_HEADER: !!process.env.MINIMAL_HEADER,
     ENTERPRISE_LEARNER_PORTAL_HOSTNAME: process.env.ENTERPRISE_LEARNER_PORTAL_HOSTNAME,
     AUTHN_MINIMAL_HEADER: !!process.env.AUTHN_MINIMAL_HEADER,
-    // this flag is introduced to unblock for new release.
-    // the flag would be removed in follow up PR
-    HIDE_USERNAME_FROM_HEADER: !!process.env.HIDE_USERNAME_FROM_HEADER,
   }, 'Header additional config');
 });
 
 const Header = ({ intl }) => {
   const { authenticatedUser, config } = useContext(AppContext);
+  const isMobile = useMediaQuery({ query: '(max-width: 769px)' });
```

Comment:
NIT: 
```suggestion
  const isDesktopScreen = useMediaQuery({ query: '(min-width: 769px)' })
```

to avoid negative conditioning.
```
 isDesktopScreen && userMenuItem,
```